keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -0,0 +1,1009 @@
+my Kubernetes pods, so that I can streamline my CI/CD pipeline. This allows me to
+maintain a small set of base images by attaching the CI/CD artifacts to them.
+Beside that, I want to package the artifacts in an OCI object to take advantage
+of OCI distribution.
+
+#### Story 3
+
+As a data scientist, MLOps engineer, or AI developer, I want to mount large language models or machine learning models in a pod alongside a model-server, so that I can efficiently serve the models
+without including them in the model-server container image. I want to package these models in an OCI object to take advantage of OCI distribution and ensure
+efficient model deployment. This allows to separate the model specifications/content from the executables that process them.
+
+#### Story 4
+
+As a security engineer, I want to use a public image for a malware scanner and
+mount in a volume of private (commercial) malware signatures, so that I can load
+those signatures without baking my own combined image (which might not be
+allowed by the copyright on the public image). Those files work regardless of
+the OS or version of the scanning software.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+- This enhancement assumes that the cluster has access to the OCI registry.
+- The implementation must handle image pull secrets and other registry authentication mechanisms.
+- Performance considerations must be taken into account, especially for large images or artifacts.
+
+### Vocabulary: OCI Images, Artifacts, and Objects
+
+**OCI Image ([spec](https://github.com/opencontainers/image-spec/blob/main/spec.md)):**
+   - A container image that conforms to the Open Container Initiative (OCI) Image Specification.
+     It includes a filesystem bundle and metadata required to run a container.
+   - Consists of multiple layers (each layer being a tarball), a manifest (which lists the layers), and a config file
+     (which provides configuration data such as environment variables, entry points, etc.).
+   - **Use Case:** Used primarily for packaging and distributing containerized applications.
+
+**OCI Artifact ([guidance](https://github.com/opencontainers/image-spec/blob/main/artifacts-guidance.md)):**
+   - An artifact describes any content that is stored and distributed using the OCI image format.
+     It includes not just container images but also other types of content like Helm charts, WASM modules, machine learning models, etc.
+   - Artifacts use the same image manifest and layer structure but may contain different types of data
+     within those layers. The artifact manifest can have media types that differ from those in standard container images.
+   - **Use Case:** Allows the distribution of non-container content using the same infrastructure and tools developed for OCI images.
+
+**OCI Object:**
+   - Umbrella term encompassing both OCI images and OCI artifacts. It represents any object that conforms to the OCI specifications for storage and distribution.
```

Comment:
Similar to comment up above, the assumption is that "all objects can be converted to a file or filesystem representation by an OCI container runtime"?  If so, that's a good framing that might belong in notes / goals (all OCI objects that can be converted to files or directories can be mounted).